pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -495,12 +476,42 @@ func getTPMapMatchingIncomingAffinityAntiAffinity(pod *v1.Pod, nodeInfoMap map[s
 				}
 			}
 		}
-		if len(nodeTopologyPairsAffinityPodsMaps.topologyPairToPods) > 0 || len(nodeTopologyPairsAntiAffinityPodsMaps.topologyPairToPods) > 0 {
-			appendResult(node.Name, nodeTopologyPairsAffinityPodsMaps, nodeTopologyPairsAntiAffinityPodsMaps)
+
+		if len(nodeTopologyPairsAffinityPodsMaps.topologyPairToPods) > 0 {
+			affinityPodsCh <- nodeTopologyPairsAffinityPodsMaps
 		}
+		if len(nodeTopologyPairsAntiAffinityPodsMaps.topologyPairToPods) > 0 {
+			antiAffinityPodsCh <- nodeTopologyPairsAntiAffinityPodsMaps
+		}
+	}
+	workqueue.ParallelizeUntil(ctx, 16, allNodesCount, processNode)
+
+	close(affinityPodsCh)
+	close(antiAffinityPodsCh)
+
+	select {
+	case err := <-errCh:
+		return nil, nil, err
+	default:
+		topologyPairsAffinityPodsMaps = newTopologyPairsMaps()
+		for value := range affinityPodsCh {
+			topologyPairsAffinityPodsMaps.appendMaps(value)
+		}
+
+		topologyPairsAntiAffinityPodsMaps = newTopologyPairsMaps()
+		for value := range antiAffinityPodsCh {
+			topologyPairsAntiAffinityPodsMaps.appendMaps(value)
+		}
+
+		return topologyPairsAffinityPodsMaps, topologyPairsAntiAffinityPodsMaps, nil
+	}
+}
+
+func catchError(errCh chan<- error, err error) {
```

Comment:
Can we move this function to [utils.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/util/utils.go) file so that we can use it in other places, and name it ```NonBlockingSendError```.